### PR TITLE
Implemented Collectables

### PR DESCRIPTION
**Pull request by rdg922**
This adds a collectable class to be inherited from. They need some network magic from you to make sure they're synced for clients that join late, and also that they're deleted on all clients. 

The collectable class means you have to override the parent _on_collect() function to do whatever you want and maybe change the export variable time_until_flashing, and you're good to go for making a new collectable that will automatically disappear after a set amount of time unless collected.

### CODE REVIEW

File: engine/collectable.gd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	timer.one_shot = true
+
+func _flash():
+	flash_count+=1
+	if(flash_count == TOTAL_FLASH_COUNT * 2):
+		queue_free()
+
+	if($Sprite.visible):
+		$Timer.wait_time = FLASH_TIME_NOT_VISIBLE
+	if(!$Sprite.visible):
```
Comment (josephbmanley) [inline engine/collectable.gd:33]:
```
if($Sprite.visible):
  [Do stuff one]
else:
  [Do stuff two]
```

File: tiles./tall_grass.gd [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+func enter_cut_cells(value):
+	cut_cells = value
+	for cell in cut_cells:
+		set_cellv(cell, -1)
+
+func process_tile(tile):
+	if get_cellv(tile) == -1:
+		return
+	cut_cells.append(tile)
+	set_cellv(tile, -1)
```
Comment (josephbmanley) [inline tiles./tall_grass.gd:24]: I suggest a comment in this area. I'm not 100% super what this function processes.

File: tiles/tall_grass.gd [hunk 3]
```diff
@@ -26,3 +26,8 @@ func process_tile(tile):
 	var grass_cut = preload("res://effects/grass_cut.tscn").instance()
 	network.current_map.add_child(grass_cut)
 	grass_cut.global_position = map_to_world(tile) + Vector2(8,6)
+
```
Comment (josephbmanley) [inline tiles/tall_grass.gd:29]: What's the difference between ` tiles./tall_grass.gd` and `tiles/tall_grass.gd`?

It looks like you make a duplicate.

**Review by josephbmanley — DISMISSED** [review]
I made a few comments.

File: tiles/tall_grass.gd (hunk 3 above)
Comment (rdg922) [reply tiles/tall_grass.gd:29]: Yep, duplicate is deleted now

**Pull request closed without merging**